src/main/java/org/rumbledb/compiler/TranslationVisitor.java

```diff
@@ -266,9 +277,22 @@ public Node visitFlowrExpr(JsoniqParser.FlowrExprContext ctx) {
                         createMetadataFromContext(ctx)
                 );
             }
-            childClause.setPreviousClause(previousFLWORClause);
-            previousFLWORClause = childClause;
+
+            // there might have been multiple variables in the syntactic starting clause
+            // and each one is mapped to a separate start clause on the logical level, so we need to rewind
+            // and populate content clauses appropriately
+            clausesTemp.clear();
+            Clause lastChildClause = childClause;
+            while (childClause.getPreviousClause() != null) {
+                clausesTemp.add(childClause);
+                childClause = childClause.getPreviousClause();
+            }
```

Comment:
The code here seems duplicated from above. Would it be possible to extract it to a method which I am hoping would increase readability as well. Also the comments seem to be identical, is this intentional or should some editing take place there? Lastly, I believe multiple variables can happen in any for and let variables and not only in the starting clauses. I was worrying if the current statement can be misleading.

Reply:
This is because it is only temporary. This code is already removed in the next PR (let clause)